lib/ansible/executor/task_executor.py

```diff
@@ -684,8 +684,9 @@ def _execute(self, variables=None):
                 self._handler.cleanup()
             display.debug("handler run complete")
 
-            # preserve no log
-            result["_ansible_no_log"] = no_log
+            # propagate no log to result- the action can set this, so only overwrite it with the task's value if missing or falsey
+            if not result.get('_ansible_no_log'):
+                result["_ansible_no_log"] = no_log
```

Comment:
Should we compare both and only disable no_log if both agree? this could lead to no_log=true in task and no_log=false from result, disclosing.

Reply:
```result["_ansible_no_log"] = no_log or result.get('_ansible_no_log', False)```

Reply:
Read it again... :wink:

Reply:
But yeah, the ternary is arguably clearer than the negation

Reply:
ah, yes, missed the `not`